src/Storage/AuthorizationCodeStorage.php

```diff
@@ -85,4 +89,29 @@ public function expireAuthorizationCode( $code ) {
 			delete_user_meta( $user_id, self::META_KEY_PREFIX . '_' . $key . '_' . $code );
 		}
 	}
+
+	/**
+	 * This function cleans up auth codes that are sitting in the database because of interrupted/abandoned OAuth flows.
+	 */
+	public function cleanupOldCodes() {
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+		$data = $wpdb->get_results( "SELECT user_id, meta_key, meta_value FROM $wpdb->usermeta WHERE meta_key LIKE 'oidc_expires_%';" );
+		if ( empty( $data ) ) {
+			return;
+		}
+
+		foreach ( $data as $row ) {
+			$expiry = absint( $row->meta_value );
+			$code   = str_replace( 'oidc_expires_', '', $row->meta_key );
+
+			// wait for an hour past expiry, to offer a chance at debug.
+			if ( time() > $expiry + 3600 ) {
```

Comment:
Of course this check now won't be necessary anymore.

Reply:
Yep!